p2p/transport/websocket/websocket.go

```diff
@@ -81,6 +83,45 @@ func WithTLSConfig(conf *tls.Config) Option {
 	}
 }
 
+// WithAllowForwardedHeader configures whether to allow the usage of
+// Forwarded and X-Forwarded-For headers to determine the real client
+// IP address when behind a proxy or load balancer.
+// configurable a list of trusted proxy IP addresses or CIDR ranges.
+// Only when a connection is from a trusted proxy, Forwarded headers will be used.
+// If trustedProxies is empty, everything is trusted.
+func WithAllowForwardedHeader(trustedProxies []string) Option {
```

Comment:
I'd accept a `[]netip.Prefix` so you get type safety and avoid all this parsing logic.